src/file.ts

```diff
@@ -2098,6 +2099,8 @@ class File extends ServiceObject<File> {
     const fileStream = this.createReadStream(options);
     let receivedData = false;
     if (destination) {
+      fs.mkdirSync(path.dirname(destination), {recursive: true});
```

Comment:
One thing that we might want to handle here is if the directory already exists. I believe in that situation this will throw an error. Additionally since this code path can be used for a single file `path.dirname('foo')` will return `'.'`.

Reply:
Thanks @ddelgrosso1 I had a look at the docs, and `fs.mkdirSync` has the same behaviour as `fs.mkDir`, in which:
- `recursive: false` (default) will throw `EEXIST`
- `recursive: true` will resolve without an error

It's mentioned here [in the docs](https://nodejs.org/api/fs.html#fsmkdirpath-options-callback) [all the way back to Node v12](https://nodejs.org/docs/latest-v12.x/api/fs.html#fs_fs_mkdir_path_options_callback). 

> Calling fs.mkdir() when path is a directory that exists results in an error only when recursive is false

Docs can't always be trusted, so I tried it out:

Recursive: true ===> no error
![recursive-true](https://github.com/googleapis/nodejs-storage/assets/12564636/70dd09a1-02db-4e3e-b933-cb9b8ca6b5ee)

Recursive: true ===> EEXIST error
![recursive-false](https://github.com/googleapis/nodejs-storage/assets/12564636/f238f062-b3fd-4b34-9f8f-9ade51337d49)

Reply:
That being said, we could add a `fs.existsSync()` just to be safe. Let me know, thanks!

Reply:
I think it is fine without the additional check (I probably should have looked at the docs first 😆)